core/src/upgrade.rs

```diff
@@ -134,6 +135,28 @@ impl<T: AsRef<[u8]>> ProtocolName for T {
     }
 }
 
+/// Wrapper for printing a [`ProtocolName`] that is expected to be mostly ASCII
+///
+/// ```rust
+/// # use libp2p_core::upgrade::DisplayProtocolName;
+/// assert_eq!(DisplayProtocolName("hello").to_string(), "hello");
+/// assert_eq!(DisplayProtocolName("hellö/").to_string(), "hell<C3><B6>/");
+/// ```
+pub struct DisplayProtocolName<N>(pub N);
+impl<N: ProtocolName> fmt::Display for DisplayProtocolName<N> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use fmt::Write;
+        for byte in self.0.protocol_name() {
+            if (b' '..=b'~').contains(byte) {
+                f.write_char(char::from(*byte))?;
+            } else {
+                write!(f, "<{:02X}>", byte)?;
```

Comment:
```suggestion
                write!(f, "0x{:02X}", byte)?;
```

Thoughts?

Reply:
We should probably use something that is established already. I went with what I know from `less`, we could also use `\xAB` like many programming languages use. Either way, making the output unambiguously parseable back to the bytes is a different story; I don’t think that would be appreciated enough to be worth it.

Reply:
I am fine to keep it as is :)

Reply:
My intention was to convey the contents without losses, since this is a debugging tool.